includes/admin/class-sensei-onboarding.php

```diff
@@ -105,6 +106,33 @@ public function admin_menu() {
 
 	}
 
+	/**
+	 * Redirect after first activation.
+	 */
+	public function activation_redirect() {
+		if (
+			// Check if activation redirect is needed.
+			! get_transient( 'sensei_activation_redirect' )
+			// Test whether the context of execution comes from async action scheduler.
+			// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Arguments used for comparison.
+			|| ( isset( $_REQUEST['action'] ) && 'as_async_request_queue_runner' === $_REQUEST['action'] )
+			// On these pages, or during these events, postpone the redirect.
+			|| wp_doing_ajax() || wp_doing_cron() || is_network_admin() || ! current_user_can( 'manage_sensei' )
+		) {
+			return;
+		}
+
+		delete_transient( 'sensei_activation_redirect' );
```

Comment:
To be safe, should this transient also be deleted when the plugin is uninstalled?

Reply:
I don't think it would be a problem, as the transient has only 30 seconds. But it doesn't hurt to put it in uninstall :)
Added here: https://github.com/Automattic/sensei/pull/3102/commits/bd1ce034295cd65e90e4ac96f7ecffdaa21e711b